src/plugins/lua/replies.lua

```diff
@@ -91,16 +98,146 @@ local function replies_check(task)
+      { exp = settings['expire'], mls = settings['max_local_size'] })
+    local_replies_set_script = lua_redis.add_redis_script(set_script_zadd_local, redis_params)
+  end
+
+  local function add_to_global_replies_set(recipients, global_key, task_time)
+    lua_util.debugm(N, task, 'Adding recipients %s to global replies set', recipients)
+
+    local function zadd_global_set_cb(err, data)
+      if err ~= nil then
+        rspamd_logger.errx(task, 'failed to add recipients %s to global replies set with error: %s', recipients, err)
+        return
+      end
+      rspamd_logger.infox(task, 'added recipients %s to global set', recipients)
+    end
+    for _, rcpt in ipairs(recipients) do
+      lua_redis.exec_redis_script(global_replies_set_script,
+              { task=task, is_write = true },
+              zadd_global_set_cb,
+              { global_key },
+              { task_time, rcpt })
+    end
+  end
+
+  local function update_global_replies_set(recipients, sender_key, global_key, task_time)
+    local function redis_zrange_cb(err, data)
+      if err ~= nil then
+        rspamd_logger.errx(task,
+                'redis_zrange_cb error when reading zrange withscores from global replies set with error: %s', err)
+        return
+      end
+      local last_score = tonumber(data[#data])
+      lua_util.debugm(N, task, 'last score %s of global replies set was received', last_score)
+
+      -- if last score wasn't found
+      if last_score == nil then
+        lua_util.debugm(N, task,
+                'have not found any senders in global replies set, considering last score as 0')
+        last_score = 0
+      end
+
+      -- updating params considering last score of existing sender
+      task_time = task_time + last_score
+
+      add_to_global_replies_set(recipients, global_key, tostring(task_time))
+    end
+
+    lua_util.debugm(N, task, 'Getting recipients withscores from global replies set to get last score')
+
+    -- getting scores of recipients in global replies set
+    lua_redis.redis_make_request(task,
+            redis_params,
+            sender_key,
+            false,
+            redis_zrange_cb,
+            'ZRANGE',
+            {global_key, '-1', '-1', 'WITHSCORES'}
+    )
+  end
+
+  local function add_to_replies_set(recipients)
+    local sender = task:get_reply_sender()
+
+    local task_time = task:get_timeval(true)
+
+    -- making params out of recipients list for replies set
+    local task_time_str = tostring(task_time)
+
+    local sender_string = lua_util.maybe_obfuscate_string(tostring(sender), settings, settings.sender_prefix)
+    local sender_key = make_key(sender_string:lower(), 8)
+
+    lua_util.debugm(N, task,
+            'Adding recipients %s to sender %s local replies set', recipients, sender_key)
+
+    local global_key = make_key(settings.sender_key_global, settings.sender_key_size, settings.sender_prefix)
+
+    configure_redis_scripts()
```

Comment:
Could we move this out of symbol callback similar to how `ratelimit` module is doing it?